src/main/java/mateacademy/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -24,19 +25,15 @@ public class UserServiceImpl implements UserService {
     @Override
     public UserResponseDto register(RegisterUserRequestDto userRequestDto)
             throws RegistrationException {
-        if (userRepository.findByEmail(userRequestDto.getEmail()).isPresent()) {
-            throw new RegistrationException("Can't register this user");
-        }
-        if (checkPassword(userRequestDto.getPassword(), userRequestDto.getRepeatPassword())) {
+        if (userRequestDto.getPassword().equals(userRequestDto.getRepeatPassword())) {
+            if (userRepository.findByEmail(userRequestDto.getEmail()).isPresent()) {
+                throw new RegistrationException("Can't register this user");
+            }
```

Comment:
You can combine these conditional but it is better to have different messages for both cases.